### PR TITLE
Fixing float to string not setting null count

### PR DESCRIPTION
**Pull request by hyperbolic2346**
This is a fix from the fallout of the cudf change to not allow unknown validity. We computed, incorrectly, the null count in this kernel, but then didn't use that value at all. I've updated it to properly compute the validity and actually store it in the column. This isn't the fastest approach, but it works. We can investigate performance later if it becomes an issue.

### CODE REVIEW

File: src/main/cpp/src/cast_string_to_float.cu [hunk 1]
```diff
@@ -555,22 +555,14 @@ class string_to_float {
...
-      if (_ansi_except && except) { atomicMax(_ansi_except, _num_rows - _row); }
-    }
+    if (threadIdx.x == 0 && _ansi_except && except) { atomicMax(_ansi_except, _num_rows - _row); }
 
     // 0th thread in each warp updates the validity
     size_type const row_id = _warp_id;
     if (threadIdx.x % 32 == 0 && valid) {
       // uses atomics
       cudf::set_bit(_validity, row_id);
+      atomicAdd(_valid_count, 1);
```
Comment (mythrocks) [inline src/main/cpp/src/cast_string_to_float.cu:565]: Ah, ok. This makes sense, phrased this way.

**Review by mythrocks — APPROVED** [review]
LGTM! The sync-test now runs cleanly.

**Comment by pxLi** [comment]
@hyperbolic2346 can you sign the commit? thanks~

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by nvdbaranec** [comment]
An alternative solution to consider:  don't have `compute_validity` compute null count.  Just have it update the validity vector.  Instead, have the operator() itself return 1 or 0 for null/non null and then do one big `single_lane_block_sum_reduce()` and the end of the kernel and store the null count once, without the atomic.

**Pull request merged**